cookbook/security/voters.rst

```diff
@@ -35,120 +35,179 @@ The Voter Interface
     }
 
-In this example, the voter will check if the user has access to a specific
-object according to your custom conditions (e.g. they must be the owner of
-the object). If the condition fails, you'll return
-``VoterInterface::ACCESS_DENIED``, otherwise you'll return
-``VoterInterface::ACCESS_GRANTED``. In case the responsibility for this decision
-does not belong to this voter, it will return ``VoterInterface::ACCESS_ABSTAIN``.
+.. versionadded::
+    The ``Voter`` helper class was added in Symfony 2.8. In earlier versions, an
+    ``AbstractVoter`` class with similar behavior was available.
+
+.. _how-to-use-the-voter-in-a-controller:
+
+Setup: Checking for Access in a Controller
+------------------------------------------
+
+Suppose you have a ``Post`` object and you need to decide whether or not the current
+user can *edit* or *view* the object. In your controller, you'll check access with
+code like this::
+
+    // src/AppBundle/Controller/PostController.php
+    // ...
+
+    class PostController extends Controller
+    {
+        /**
+         * @Route("/posts/{id}", name="post_show")
+         */
+        public function showAction($id)
+        {
+            // get a Post object - e.g. query for it
+            $post = ...;
+
+            // check for "view" access: calls all voters
+            $this->denyAccessUnlessGranted('view', $post);
+
+            // ...
+        }
+
+        /**
+         * @Route("/posts/{id}/edit", name="post_edit")
+         */
+        public function editAction($id)
+        {
+            // get a Post object - e.g. query for it
+            $post = ...;
+
+            // check for "edit" access: calls all voters
+            $this->denyAccessUnlessGranted('edit', $post);
+
+            // ...
+        }
+    }
+
+The ``denyAccessUnlessGranted()`` method (and also, the simpler ``isGranted()`` method)
+calls out to the "voter" system. Right now, no voters will vote on whether or not
+the user can "view" or "edit" a ``Post``. But you can create your *own* voter that
+decides this using whatever logic you want.
+
+.. tip::
+
+    The ``denyAccessUnlessGranted()`` function and the ``isGranted()`` functions
+    are both just shortcuts to call ``isGranted()`` on the ``security.authorization_checker``
+    service.
 
 Creating the custom Voter
 -------------------------
 
-The goal is to create a voter that checks if a user has access to view or
-edit a particular object. Here's an example implementation:
+Suppose the logic to decide if a user can "view" or "edit" a ``Post`` object is
+pretty complex. For example, a ``User`` can always edit or view a ``Post`` they created.
+And if a ``Post`` is marked as "public", anyone can view it. A voter for this situation
+would look like this::
 
-.. code-block:: php
-
-    // src/AppBundle/Security/Authorization/Voter/PostVoter.php
-    namespace AppBundle\Security\Authorization\Voter;
+    // src/AppBundle/Security/PostVoter.php
+    namespace AppBundle\Security;
```

Comment:
Why changing the namespace to use `AppBundle\Security` instead of `AppBundle\Security\Authorization\Voter` ?

If the application grows, it could have many things in the `Security` namespace. I think the cookbooks are good indications for a new developper to know where to place common things and promote some standard directory structures.